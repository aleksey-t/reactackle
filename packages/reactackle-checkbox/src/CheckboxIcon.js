import React from 'react';
import PropTypes from 'prop-types';
import { withTooltip } from 'reactackle-tooltip';
import { withTheme } from 'reactackle-core';
import { CheckboxIconStyled } from './styles/CheckboxIconStyled';

const propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  tooltipText: PropTypes.string,
  theme: PropTypes.object.isRequired,
  Tooltip: PropTypes.func.isRequired,
  toggleTooltip: PropTypes.func.isRequired,
  showTooltip: PropTypes.func.isRequired,
  hideTooltip: PropTypes.func.isRequired,
};

const defaultProps = {
  checked: void 0,
  disabled: false,
  tooltipText: '',
};

const CheckboxIconComponent = props => {
  const Tooltip = props.Tooltip;

  const tooltip = props.tooltipText
    ? <Tooltip text={props.tooltipText} />
    : null;

  const tooltipManagementProps = props.tooltipText
    ? {
        onClick: props.toggleTooltip,
        onFocus: props.showTooltip,
        onBlur: props.hideTooltip,
        onMouseEnter: props.showTooltip,
        onMouseLeave: props.hideTooltip,
        tabIndex: 1,
      }
    : {};

  return (
    <CheckboxIconStyled
      checked={props.checked}
      disabled={props.disabled}
      {...tooltipManagementProps}
    >
      {props.theme.reactackle.components.checkbox.icon.src}
      {tooltip}
    </CheckboxIconStyled>
  );
};

CheckboxIconComponent.propTypes = propTypes;
CheckboxIconComponent.defaultProps = defaultProps;
CheckboxIconComponent.displayName = 'CheckboxIcon';

export default withTheme(withTooltip(CheckboxIconComponent, true));
