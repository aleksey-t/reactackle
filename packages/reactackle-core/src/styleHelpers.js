/** Base Module Calculations */
export const baseModule = ({ baseModule }) => baseModule;
export const halfBaseModule = ({ baseModule }) => baseModule * 0.5;
export const oneAndHalfBaseModule = ({ baseModule }) => baseModule * 1.5;
export const doubleBaseModule = ({ baseModule }) => baseModule * 2;
export const twoAndHalfBaseModule = ({ baseModule }) => baseModule * 2.5;
export const tripleBaseModule = ({ baseModule }) => baseModule * 3;

const multiplierFunctions = Object.create(null);

export const bmodule = multiplier => {
  if (!multiplierFunctions[multiplier]) {
    multiplierFunctions[multiplier] =
      ({ baseModule }) => baseModule * multiplier;
  }

  return multiplierFunctions[multiplier];
};

/** Size parameters */
export const contentWidth = ({ contentWidth }) => contentWidth;
export const totalColumns = ({ totalColumns }) => totalColumns;

export const radiusDefault = ({ radiusDefault }) => radiusDefault;
export const radiusRounded = ({ radiusRounded }) => radiusRounded;

/** Breakpoints */
export const breakpointXsmall = ({ breakpoints }) => breakpoints.xsmall;
export const breakpointSmall = ({ breakpoints }) => breakpoints.small;
export const breakpointMedium = ({ breakpoints }) => breakpoints.medium;
export const breakpointLarge = ({ breakpoints }) => breakpoints.large;
export const breakpointXlarge = ({ breakpoints }) => breakpoints.xlarge;
export const breakpointXxlarge = ({ breakpoints }) => breakpoints.xxlarge;

/** Colors */
export const colorWhite = ({ color }) => color.white;
export const colorBlack = ({ color }) => color.black;
export const colorTransparent = ({ color }) => color.transparent;

export const colorMain = ({ color }) => color.main;
export const colorMainForeground = ({ color }) => color.mainFgTextColor;
export const colorSecondary = ({ color }) => color.secondary;
export const colorSecondaryForeground = ({ color }) =>
  color.secondaryFgTextColor;
export const colorAlert = ({ color }) => color.alert;
export const colorAlertForeground = ({ color }) => color.alertFgTextColor;
export const colorWarning = ({ color }) => color.warning;
export const colorWarningForeground = ({ color }) => color.warningFgTextColor;
export const colorError = ({ color }) => color.alert;
export const colorErrorForeground = ({ color }) => color.alertFgTextColor;
export const colorSuccess = ({ color }) => color.success;
export const colorSuccessForeground = ({ color }) => color.successFgTextColor;
export const colorInfo = ({ color }) => color.info;
export const colorInfoForeground = ({ color }) => color.infoFgTextColor;

export const colorPaletteGrey50 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[50];
export const colorPaletteGrey75 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[75];
export const colorPaletteGrey100 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[100];
export const colorPaletteGrey200 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[200];
export const colorPaletteGrey300 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[300];
export const colorPaletteGrey400 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[400];
export const colorPaletteGrey500 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[500];
export const colorPaletteGrey600 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[600];
export const colorPaletteGrey700 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[700];
export const colorPaletteGrey800 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[800];
export const colorPaletteGrey900 = ({ colorPaletteGrey }) =>
  colorPaletteGrey[900];

export const colorBorder = ({ colorBorder }) => colorBorder;
export const colorAnchor = ({ colorAnchor }) => colorAnchor;

export const colorBrandFacebook = ({ colorBrand }) => colorBrand.facebook;
export const colorBrandVkontakte = ({ colorBrand }) => colorBrand.vkontakte;
export const colorBrandGooglePlus = ({ colorBrand }) => colorBrand.googlePlus;

/** Font */
export const fontFamilySansSerif = ({ fontFamily }) => fontFamily.sansSerif;
export const fontFamilySerif = ({ fontFamily }) => fontFamily.serif;
export const fontFamilyMonospace = ({ fontFamily }) => fontFamily.monospace;

export const fontWeightLight = ({ fontWeight }) => fontWeight.light;
export const fontWeightNormal = ({ fontWeight }) => fontWeight.normal;
export const fontWeightSemibold = ({ fontWeight }) => fontWeight.semibold;
export const fontWeightBold = ({ fontWeight }) => fontWeight.bold;

export const fontSizeXsmall = ({ fontSize }) => fontSize[-2];
export const fontSizeSmall = ({ fontSize }) => fontSize[-1];
export const fontSizeBody = ({ fontSize }) => fontSize[0];
export const fontSizeBody2 = ({ fontSize }) => fontSize[1];
export const fontSizeTitle = ({ fontSize }) => fontSize[2];
export const fontSizeHeadline = ({ fontSize }) => fontSize[3];
export const fontSizeDisplay1 = ({ fontSize }) => fontSize[4];
export const fontSizeDisplay2 = ({ fontSize }) => fontSize[5];
export const fontSizeDisplay3 = ({ fontSize }) => fontSize[6];
export const fontSizeDisplay4 = ({ fontSize }) => fontSize[7];

export const fontColorLight = ({ fontColor }) => fontColor.light;
export const fontColorMedium = ({ fontColor }) => fontColor.medium;

/** Other */
export const directionText = ({ direction }) => direction.text;
export const directionFloat = ({ direction }) => direction.float;

export const zIndexBase = ({ zIndex }) => zIndex.base;
export const zIndexNav = ({ zIndex }) => zIndex.nav;
export const zIndexAlertMobile = ({ zIndex }) => zIndex.alertMobile;
export const zIndexDropdown = ({ zIndex }) => zIndex.dropdown;
export const zIndexOverlay = ({ zIndex }) => zIndex.overlay;
export const zIndexModal = ({ zIndex }) => zIndex.modal;
export const zIndexPopup = ({ zIndex }) => zIndex.popup;
export const zIndexAlertDesktop = ({ zIndex }) => zIndex.alertDesktop;
export const zIndexTooltip = ({ zIndex }) => zIndex.tooltip;
export const zIndexMoon = ({ zIndex }) => zIndex.moon;

/** Body Settings */
export const bodyBackgroundColor = ({ body }) => body.bgColor;
export const bodyFontColor = ({ body }) => body.fontColor;
export const bodyFontColorAlt = ({ body }) => body.fontColorAlt;
export const bodyFontFamily = ({ fontFamily }) => fontFamily.sansSerif;
export const bodyFontWeight = ({ body }) => body.fontWeight;
export const bodyLineHeight = ({ body }) => body.lineHeight;
