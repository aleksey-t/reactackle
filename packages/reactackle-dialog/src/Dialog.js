'use strict';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Portal from 'react-portal';
import {
  withTheme,
  noop,
  getKey,
  registerDefaultComponentTheme,
} from 'reactackle-core';
import { DialogHeading } from './DialogHeading';
import { DialogCloseButton } from './DialogCloseButton';
import { DialogContent } from './DialogContent';
import { DialogButton } from './DialogButton';
import { DialogActions, DialogActionsRegion } from './DialogActions';
import { ModalStyled } from './styles/ModalStyled';
import { BackdropStyled } from './styles/BackdropStyled';
import { OverflowWrapperStyled } from './styles/OverflowWrapperStyled';
import { RegionMainStyled } from './styles/RegionMainStyled';
import { RegionSideStyled } from './styles/RegionSideStyled';
import { ContentWrapperStyled } from './styles/ContentWrapperStyled';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('dialog', componentTheme);

const ButtonType = PropTypes.shape({
  ...DialogButton.propTypes,
  id: PropTypes.number,
});

const propTypes = {
  /**
   * Determine whether to render backdrop
   */
  backdrop: PropTypes.bool,
  /**
   * If set to true, the content of the Dialog will be scrollable
   */
  scrollable: PropTypes.bool,
  /**
   * Determine whether to render dialog
   */
  open: PropTypes.bool,
  /**
   * Determine whether to render close button
   */
  haveCloseButton: PropTypes.bool,
  /**
   * Determine whether to close dialog on ESC key down
   */
  closeOnEscape: PropTypes.bool,
  /**
   * Determine whether to close dialog on backdrop click
   */
  closeOnBackdropClick: PropTypes.bool,
  /**
   * Title of the dialog
   */
  title: PropTypes.string,
  /**
   * Subtitle of the dialog
   */
  subtitle: PropTypes.string,
  /**
   * Buttons of the left side of the dialog
   */
  buttonsLeft: PropTypes.arrayOf(ButtonType),
  /**
   * Buttons of the right side of the dialog
   */
  buttons: PropTypes.arrayOf(ButtonType),
  /**
   * Minimum width of the dialog
   */
  minWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /**
   * Padding size in the dialog
   */
  paddingSize: PropTypes.oneOf(['none', 'default']),
  /**
   * Determine whether to render dialog in full window size
   */
  fullWindowSize: PropTypes.bool,
  /**
   * Determine whether to render dialog with transparent background
   */
  transparentBg: PropTypes.bool,
  /**
   * Determine whether to display dialog content in flex mode
   */
  dialogContentFlex: PropTypes.bool,
  /**
   * Dialog aside element
   */
  regionAside: PropTypes.element,
  /**
   * Dialog aside element max width
   */
  regionAsideMaxWidth: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  /**
   * Determine whether to show dialog aside element in dialog
   */
  regionAsideShow: PropTypes.bool,
  /**
   * @ignore
   */
  // eslint-disable-next-line react/require-default-props
  theme: PropTypes.object,
  /**
   * Specify function to call on ENTER key press
   */
  onEnterKeyPress: PropTypes.func,
  /**
   * Specify function to call on dialog close
   */
  onClose: PropTypes.func,
  /**
   * Specify function to call just before dialog close
   */
  onBeforeClose: PropTypes.func,
  /**
   * Determines whether to display already mounted dialog
   */
  visible: PropTypes.bool,
};

const defaultProps = {
  backdrop: false,
  scrollable: false,
  open: false,
  haveCloseButton: false,
  closeOnEscape: false,
  closeOnBackdropClick: false,
  title: '',
  subtitle: '',
  buttons: [],
  buttonsLeft: [],
  minWidth: 0,
  paddingSize: 'default',
  fullWindowSize: false,
  transparentBg: false,
  dialogContentFlex: false,
  regionAside: null,
  regionAsideMaxWidth: 150,
  regionAsideShow: false,
  onEnterKeyPress: noop,
  onClose: noop,
  onBeforeClose: noop,
  visible: true,
};

const ESC = 27,
  ENTER = 13;

class _Dialog extends Component {
  constructor(props) {
    super(props);

    this.canHaveBackdrop = true;
    this._keyboardListened = false;
    this._backupBodyStyle = {
      height: null,
      overflowY: null,
    };

    this._handleBackdropClick = this._handleBackdropClick.bind(this);
    this._handleKeyUp = this._handleKeyUp.bind(this);
    this._handleDialogClose = this._handleDialogClose.bind(this);
  }

  componentDidMount() {
    this._toggleKeyboardListener(this._isVisible());
  }

  componentWillReceiveProps(nextProps) {
    this._toggleKeyboardListener(this._isVisible(nextProps));
  }

  componentWillUnmount() {
    this._toggleKeyboardListener(false);
  }


  _toggleKeyboardListener(condition) {
    if (condition && !this._keyboardListened) {
      this._keyboardListened = true;
      document.body.addEventListener('keyup', this._handleKeyUp);
    } else if (!condition && this._keyboardListened) {
      this._keyboardListened = false;
      document.body.removeEventListener('keyup', this._handleKeyUp);
    }
  }

  _isVisible(source = this.props) {
    return source.open && source.visible;
  }

  _handleDialogClose() {
    const willNotClose = this.props.onBeforeClose() === false;

    if (!willNotClose) this.props.onClose();
  }

  _handleBackdropClick() {
    if (this.props.closeOnBackdropClick) this._handleDialogClose();
  }

  _handleKeyUp(event) {
    if (event.keyCode === ESC && this.props.closeOnEscape)
      this._handleDialogClose();
    else if (event.keyCode === ENTER)
      this.props.onEnterKeyPress(this._handleDialogClose);
  }

  _renderCloseButton() {
    if (!this.props.haveCloseButton) return null;

    return (
      <DialogCloseButton
        onClick={this._handleDialogClose}
        transparentBg={this.props.transparentBg}
      />
    );
  }

  _renderRegionAside() {
    if (!this.props.regionAside) return null;

    return (
      <RegionSideStyled regionAsideMaxWidth={this.props.regionAsideMaxWidth}>
        {this.props.regionAside}
      </RegionSideStyled>
    );
  }

  _renderButtonsLeft() {
    if (!this.props.buttonsLeft.length) return null;

    const buttonsLeftList = this.props.buttonsLeft.map((button, index) =>
      <DialogButton
        {...button}
        key={getKey(button, index)}
        onCloseDialog={this._handleDialogClose}
      />,
    );

    return (
      <DialogActionsRegion alignLeft>
        {buttonsLeftList}
      </DialogActionsRegion>
    );
  }

  _renderButtons() {
    if (!this.props.buttons.length) return null;

    const buttonsList = this.props.buttons.map((button, index) =>
      <DialogButton
        {...button}
        key={getKey(button, index)}
        onCloseDialog={this._handleDialogClose}
      />,
    );

    return (
      <DialogActionsRegion>
        {buttonsList}
      </DialogActionsRegion>
    );
  }

  _renderActions() {
    if (!this.props.buttons.length && !this.props.buttonsLeft.length)
      return null;

    const buttonsLeft = this._renderButtonsLeft(),
      buttons = this._renderButtons();

    return (
      <DialogActions>
        {buttonsLeft}
        {buttons}
      </DialogActions>
    );
  }

  _renderBackdrop() {
    if (!this.canHaveBackdrop || !this.props.backdrop) return null;

    return <BackdropStyled onClick={this._handleBackdropClick} />;
  }

  render() {
    const closeButton = this._renderCloseButton(),
      regionAside = this._renderRegionAside(),
      actions = this._renderActions(),
      backdrop = this._renderBackdrop();

    const followsHeading = !!this.props.title || !!this.props.subtitle,
      hasRegionAside = this.props.regionAside !== null;

    return (
      <Portal isOpened={this.props.open}>
        <ModalStyled visible={this.props.visible}>
          {backdrop}
          <OverflowWrapperStyled
            customMinWidth={this.props.minWidth}
            hasRegionAside={hasRegionAside}
            regionAsideMaxWidth={this.props.regionAsideMaxWidth}
            regionAsideShow={this.props.regionAsideShow}
          >
            <ContentWrapperStyled
              paddingSize={this.props.paddingSize}
              fullWindowSize={this.props.fullWindowSize}
              customMinWidth={this.props.minWidth}
              hasTransparentBg={this.props.transparentBg}
              hasRegionAside={hasRegionAside}
              regionAsideMaxWidth={this.props.regionAsideMaxWidth}
              regionAsideShow={this.props.regionAsideShow}
            >
              <RegionMainStyled scrollable={this.props.scrollable}>
                {closeButton}
                <DialogHeading
                  title={this.props.title}
                  subtitle={this.props.subtitle}
                  paddingSize={this.props.paddingSize}
                  haveCloseButton={this.props.haveCloseButton}
                />
                <DialogContent
                  flex={this.props.dialogContentFlex}
                  paddingSize={this.props.paddingSize}
                  scrollable={this.props.scrollable}
                  followsHeading={followsHeading}
                >
                  {this.props.children}
                </DialogContent>
                {actions}
              </RegionMainStyled>
              {regionAside}
            </ContentWrapperStyled>
          </OverflowWrapperStyled>
        </ModalStyled>
      </Portal>
    );
  }
}

_Dialog.propTypes = propTypes;
_Dialog.defaultProps = defaultProps;
_Dialog.displayName = 'Dialog';

export default withTheme(_Dialog);
