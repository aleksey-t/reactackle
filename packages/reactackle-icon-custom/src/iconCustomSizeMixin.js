import { css } from 'styled-components';

export const iconCustomSizeMixin = (
  height,
  imgSize = height,
  width = height,
) => css`
  width: ${width};
  height: ${height};
  line-height: ${height} !important;
  background-size: ${imgSize};
`;
