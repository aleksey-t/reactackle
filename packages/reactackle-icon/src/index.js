import Icon from './Icon';

export { Icon };
export * from './iconPropType';
export * from './iconStyleMixin';
