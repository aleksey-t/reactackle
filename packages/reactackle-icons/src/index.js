import { registerDefaultComponentTheme } from 'reactackle-core';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('icon', componentTheme);

export * from './IconCustom';
export * from './IconSvg';
export * from './iconStyleMixin';
export * from './iconsPropType';
