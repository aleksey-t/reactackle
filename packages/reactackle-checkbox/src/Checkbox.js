import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { noop, isUndef, registerDefaultComponentTheme } from 'reactackle-core';
import { TooltipIcon } from 'reactackle-tooltip-icon';
import { CheckboxStyled } from './styles/CheckboxStyled';
import { CheckboxElementStyled } from './styles/CheckboxElementStyled';
import { CheckboxInputStyled } from './styles/CheckboxInputStyled';
import { CheckboxLabelStyled } from './styles/CheckboxLabelStyled';
import { LabelElementStyled } from './styles/LabelElementStyled';
import { LabelTextStyled } from './styles/LabelTextStyled';
import { MessageStyled } from './styles/MessageStyled';
import CheckboxIcon from './CheckboxIcon';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('checkbox', componentTheme);

const propTypes = {
  /**
   * Determines whether to check Checkbox by default
   */
  defaultChecked: PropTypes.bool,
  /**
   * Determines Checkbox checked state
   */
  checked: PropTypes.bool,
  /**
   * Determines whether to disable checkbox
   */
  disabled: PropTypes.bool,
  /**
   * Label of Checkbox
   */
  label: PropTypes.string,
  /**
   * Label Component
   */
  labelElement: PropTypes.element,
  /**
   * Add tooltip
   */
  tooltip: PropTypes.string,
  /**
   * Pass error message
   */
  errorMessage: PropTypes.string,
  /**
   * Specify function that will be called on Checkbox checked state change
   */
  onChange: PropTypes.func,
};

const defaultProps = {
  defaultChecked: false,
  checked: void 0,
  disabled: false,
  label: '',
  tooltip: '',
  errorMessage: '',
  labelElement: null,
  onChange: noop,
};

let nextId = 0;

export default class Checkbox extends Component {
  constructor(props) {
    super(props);

    this._inputId = `checkbox-${nextId++}`;

    this.state = {
      checked: isUndef(props.checked) ? props.defaultChecked : props.checked,
    };

    this._handleChange = this._handleChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (
      nextProps.checked !== this.props.checked &&
      isUndef(nextProps.checked)
    ) {
      this.setState({
        checked: this.props.checked,
      });
    }
  }

  _handleChange(event) {
    const checked = event.target.checked;

    if (isUndef(this.props.checked)) {
      this.setState({
        checked,
      });
    }

    this.props.onChange({
      value: checked,
    });
  }

  render() {
    const checked = isUndef(this.props.checked)
      ? this.state.checked
      : this.props.checked;

    const tooltip =
      this.props.tooltip && (this.props.label || this.props.labelElement)
        ? <TooltipIcon text={this.props.tooltip} />
        : null;

    const label = this.props.label
      ? <span>
          <LabelTextStyled hasTooltip={!!this.props.tooltip}>
            {this.props.label}
          </LabelTextStyled>
          {tooltip}
        </span>
      : null;

    const labelElement = this.props.labelElement
      ? <LabelElementStyled hasTooltip={!!this.props.tooltip}>
          {this.props.labelElement}
          {tooltip &&
            <LabelTextStyled>
              {tooltip}
            </LabelTextStyled>}
        </LabelElementStyled>
      : null;

    const checkboxIcon =
      !this.props.label && !this.props.labelElement
        ? <CheckboxIcon
            disabled={this.props.disabled}
            checked={checked}
            tooltipText={this.props.tooltip}
          />
        : <CheckboxIcon
            disabled={this.props.disabled}
            checked={checked}
          />;

    const message = this.props.errorMessage
      ? <MessageStyled>
          {this.props.errorMessage}
        </MessageStyled>
      : null;

    return (
      <CheckboxStyled>
        <CheckboxElementStyled
          textFree={!this.props.label && !this.props.labelElement}
        >
          <CheckboxInputStyled
            id={this._inputId}
            type="checkbox"
            checked={checked}
            disabled={this.props.disabled}
            onChange={this._handleChange}
          />
          <CheckboxLabelStyled
            textFree={!this.props.label && !this.props.labelElement}
            disabled={this.props.disabled}
            checked={checked}
            htmlFor={this._inputId}
          >
            {checkboxIcon}
            {label}
            {labelElement}
          </CheckboxLabelStyled>
        </CheckboxElementStyled>
        {message}
      </CheckboxStyled>
    );
  }
}

Checkbox.propTypes = propTypes;
Checkbox.defaultProps = defaultProps;
Checkbox.displayName = 'Checkbox';
