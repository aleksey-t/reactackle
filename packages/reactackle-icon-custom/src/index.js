import { iconCustomSizeMixin } from './iconCustomSizeMixin';
import { IconCustom } from './IconCustom';

export {
  IconCustom,
  iconCustomSizeMixin,
};
