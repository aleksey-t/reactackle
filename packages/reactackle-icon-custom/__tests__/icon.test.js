'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import { IconCustom, iconCustomSizeMixin } from '../src';

jest.mock('react-dom');

const TestWrapper = styled.div`
  ${iconCustomSizeMixin('70px', '42px', '50px')}
`;

describe('<IconCustom/>', () => {
  test('renders correctly with default props', () => {
    const tree = renderer.create(
      <IconCustom />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with iconSrc prop', () => {
    const tree = renderer.create(
      <IconCustom iconSrc="http://files.gamebanana.com/img/ico/sprays/51cb98f9d3747.png" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with border prop', () => {
    const tree = renderer.create(
      <IconCustom border />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props border and borderWidth', () => {
    const tree = renderer.create(
      <IconCustom border borderWidth={2} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props border and rounded', () => {
    const tree = renderer.create(
      <IconCustom border rounded />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props border, borderWidth, rounded', () => {
    const tree = renderer.create(
      <IconCustom border rounded borderWidth={2} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop size=custom', () => {
    const tree = renderer.create(
      <TestWrapper>
        <IconCustom size="custom" />,
      </TestWrapper>,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop size=small', () => {
    const tree = renderer.create(
      <IconCustom size="small" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop size=large', () => {
    const tree = renderer.create(
      <IconCustom size="large" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop size=xlarge', () => {
    const tree = renderer.create(
      <IconCustom size="xlarge" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop colorScheme=light', () => {
    const tree = renderer.create(
      <IconCustom colorScheme="light" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop color', () => {
    const tree = renderer.create(
      <IconCustom color="#02b4d4" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop backgroundColor', () => {
    const tree = renderer.create(
      <IconCustom backgroundColor="#e81f82" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props backgroundColor and color', () => {
    const tree = renderer.create(
      <IconCustom backgroundColor="#e81f82" color="#02b4d4" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop flip=horizontal', () => {
    const tree = renderer.create(
      <IconCustom flip="horizontal" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop flip=vertical', () => {
    const tree = renderer.create(
      <IconCustom flip="vertical" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop rotate', () => {
    const tree = renderer.create(
      <IconCustom rotate={45} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
