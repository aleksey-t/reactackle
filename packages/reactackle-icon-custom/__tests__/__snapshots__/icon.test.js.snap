// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconCustom/> renders correctly with border prop 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue gubCwH"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with default props 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue djBARX"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with iconSrc prop 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue howlSV"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop backgroundColor 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue evCsMg"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop color 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue djBARX"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop colorScheme=light 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue ixwfM"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop flip=horizontal 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue ivYBQB"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop flip=vertical 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue kISsqy"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop rotate 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue vSufI"
  color=""
  rotate={45}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop size=custom 1`] = `
<div
  className="icon-test__TestWrapper-dzMMzb ggkDKV"
>
  <div
    className="IconCustomStyled__IconCustomStyled-glxlue dkYpaH"
    color=""
    rotate={0}
    size="normal"
  />
  ,
</div>
`;

exports[`<IconCustom/> renders correctly with prop size=large 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue dfVbFt"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop size=small 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue hdCIYq"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with prop size=xlarge 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue eTpbCE"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with props backgroundColor and color 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue evCsMg"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with props border and borderWidth 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue zreGr"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with props border and rounded 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue gtzvhT"
  color=""
  rotate={0}
  size="normal"
/>
`;

exports[`<IconCustom/> renders correctly with props border, borderWidth, rounded 1`] = `
<div
  className="IconCustomStyled__IconCustomStyled-glxlue ehQynk"
  color=""
  rotate={0}
  size="normal"
/>
`;
